src/master-playlist-controller.js

```diff
@@ -689,6 +698,10 @@ export class MasterPlaylistController extends videojs.EventTarget {
       this.load();
 
       return true;
+    } else if (media &&
+        // 2) the video is a VOD
+        media.endList) {
+      this.hasPlayed_ = () => this.tech_.played().length !== 0;
```

Comment:
We might need to rethink `hasPlayed_` soon... it is complicated to get this working in all 4 cases:

* MSE & VOD
* MSE & Live
* Flash & VOD
* Flash & Live